docs/features/interceptors.md

```diff
@@ -154,14 +158,34 @@ static class Program
 {
     public static void M<T2>(T2 t)
     {
-        C.InterceptableMethod(t);
+        C.InterceptableMethod(t); // intercepts with 'Interceptor1<T2>(T2)'
     }
 }
 
 static class D
 {
     [InterceptsLocation("Program.cs", 13, 11)]
-    public static void Interceptor1(object s) => throw null!;
+    public static void Interceptor1<T3>(T3 t3) => throw null!;
+}
+```
+
+If an interceptable method signature uses type parameters from the containing type (in parameters and returns, including `this`), it won't be possible to declare an interceptor for it in another type. This is because there's no position where the original containing type's type parameters can be substituted in to the interceptor. We could revisit this limitation if needed, perhaps by reviewing the arity matching requirements proposed for *implicit and explicit extensions* as a starting point.
+
+```cs
+class C<T>
+{
+    static void Interceptable(T t) { }
```

Comment:
Nit:

```suggestion
    [Interceptable]
    static void Interceptable(T t) { }
```

Reply:
This attribute was removed from the design in #68213